src/generator/renderers/content.tsx

```diff
@@ -129,18 +132,25 @@ export async function MessageSingleASTNode({ node, context }: { node: SingleASTN
 
     case 'channel': {
       const id = node.id as string;
-      const channel = await context.callbacks.resolveChannel(id);
+      // NOTE: resolveChannel is now assumed to be sync or returns null
+      const channel = (
+        context.callbacks.resolveChannel as (
+          id: string
+        ) => { isDMBased?: () => boolean; name: string; type: ChannelType } | null
+      )?.(id);
```

Comment:
This is a bad change, as this might break existing applications. Furthermore, in a future generation of this package, I hope to remove it's dependence on discord.js and accept normal APIMessage structs, in which case, a network request may be required to retrieve channel information, which cannot happen synchronously.

Also, what's up with casting the type again here?